doc/decisions/backend_plugin.md

```diff
@@ -0,0 +1,130 @@
+# Backend Plugin
+
+## Problem
+
+- Backends store plugins in arrays which have a fixed number of slots for each plugin role. The number of plugins which can be assigned is limited,
+  making it easy to reach the limit if many plugins are in use.
+- As structs, backends are separate from the plugin interface and integrated into the core of Elektra. This makes it difficult to perform operations
+  such as nesting plugins, or to develop other implementations for backends.
+
+## Constraints
+
+- It should be possible for all existing plugins to run normally
```

Comment:
We actually change the commit/rollback semantics?

```suggestion
- It should be possible for all existing plugins to run with current get/set semantics
```

Reply:
This file is a remnant from the original #2969. I did not yet look at it's content.